File: BaiTap/Python/queue.py
class queue : 
    def __init__(self , size ):
        self.size = size ; 
        self.front = -1 ; 
        self.rear = -1 ; 
        self.Q = [None]  * size ; 
    
    def isEmpty(self):
        if (self.front == -1) : 
            return True ; 
        return False;
    
    def isFull(self):
        if (self.front == 0 and self.rear == self.size - 1 ) or (self.front == self.rear + 1 ) : 
            return True 
        return False
    
    def push(self,i):
        if self.isFull():
            print("Full")
        else:
            if self.front == -1 : 
                self.front = 0 ;
            self.rear = (self.rear + 1 ) % self.size
            self.Q[self.rear]= i 
            print("Add " + str(i) )
        
    def pop(self):
        if self.isEmpty():
            print("Empty")
            return -1 
        else : 
            element = self.Q[self.front]
            if (self.front == self.rear ):
                self.front = -1 
                self.rear =  -1 
            else :
                self.front = (self.front + 1) + self.size 
            return element

if __name__ == '__main__' : 
    q = queue(2)
    q.push(2)
    q.push(3)
    print(q.isFull())
    q.pop()
    print(q.isEmpty())
    

File: BaiTap/Java/queue.java
package Vidu;

public class queue {
	int front , rear , size ; 
	int [] Q ; 
	public queue(int size) {
		this.size = size ; 
		this.front = -1;
		this.rear = -1 ;
		this.Q = new int [size];
	}
	public boolean isEmpty() {
		return front == -1 ;
	}
	public boolean isFull() {
		return (front == 0 && rear == size - 1) || (front == rear + 1);
	}
	public void push(int i ) {
		if (isFull()) {
			System.out.println("Empty");
			return;
		}else {
			if (front == -1) {
				front = 0 ;
			}
			rear = (rear + 1 ) % size ;
			Q[rear] = i;
			System.out.println("Added to " + i);
			}
		}
	public int pop() {
		if (isEmpty()) {
			System.out.println("Empty");
			return -1;
		}else {
			int element = Q[front];
			if (front == rear ) {
				front = -1 ; 
				rear = -1  ; 
			}else {
				front = (front + 1 ) % size;
			}
			System.out.println("Pop is " + element);
			return element;
		}
	}
	public void display() {
		if (isEmpty()) {
			System.out.println("Empty");
		}else {
			int i = front ;
			while(true) {
				System.out.print(Q[i] + "->");
				if (i == rear ) {
					break;
				}
				i = (i+1) % size ; 
			}
			System.out.println("None");
		}
	}
}
